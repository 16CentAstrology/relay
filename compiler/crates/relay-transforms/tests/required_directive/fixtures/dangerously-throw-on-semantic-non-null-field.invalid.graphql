# expected-to-throw
#
# Test: DANGEROUSLY_THROW_ON_SEMANTICALLY_NULLABLE_FIELD cannot be used on semantically non-null fields
# Property: Fields marked with @semanticNonNull (e.g., composer: User @semanticNonNull) should not allow DANGEROUSLY_THROW_ON_SEMANTICALLY_NULLABLE_FIELD action
# Expected: Error suggesting to use THROW instead

query MyQuery {
  opera {
    composer @required(action: DANGEROUSLY_THROW_ON_SEMANTICALLY_NULLABLE_FIELD) {
      name
    }
  }
}
