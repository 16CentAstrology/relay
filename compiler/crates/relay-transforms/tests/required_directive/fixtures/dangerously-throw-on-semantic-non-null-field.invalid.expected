==================================== INPUT ====================================
# expected-to-throw
#
# Test: DANGEROUSLY_THROW_ON_SEMANTICALLY_NULLABLE_FIELD cannot be used on semantically non-null fields
# Property: Fields marked with @semanticNonNull (e.g., composer: User @semanticNonNull) should not allow DANGEROUSLY_THROW_ON_SEMANTICALLY_NULLABLE_FIELD action
# Expected: Error suggesting to use THROW instead

query MyQuery {
  opera {
    composer @required(action: DANGEROUSLY_THROW_ON_SEMANTICALLY_NULLABLE_FIELD) {
      name
    }
  }
}
==================================== ERROR ====================================
✖︎ @required with action DANGEROUSLY_THROW_ON_SEMANTICALLY_NULLABLE_FIELD is not allowed on non-nullable fields. Use action THROW instead.

  dangerously-throw-on-semantic-non-null-field.invalid.graphql:9:32
    8 │   opera {
    9 │     composer @required(action: DANGEROUSLY_THROW_ON_SEMANTICALLY_NULLABLE_FIELD) {
      │                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   10 │       name
